frontend/src/lib/components/CompareFilter/CompareFilter.tsx

```diff
@@ -16,13 +17,28 @@ export function CompareFilter(): JSX.Element | null {
         return null
     }
 
+    const label = (
+        <span className="font-normal">
+            <RollingDateRangeFilter
+                dateRangeFilterLabel="Compare to "
+                dateRangeFilterSuffixLabel=" earlier"
+                allowPeriod={true}
+                dateFrom={compareTo}
+                inUse={true}
+                onChange={(compareTo) => {
+                    updateInsightFilter({ compare: true, compareTo })
+                }}
+            />
```

Comment:
This current design doesn't fit in super well into the existing UI, because:
1. `CompareFilter` is currently taller than any other elements in its row
2. This `RollingDateRangeFilter` is rendered as a button, but it shouldn't. Should be just a div really, and only the select for choosing the comparison unit should be a button

But with the above addressed it should be pretty sleek!

Reply:
<img width="1018" alt="image" src="https://github.com/PostHog/posthog/assets/1855120/8c68c6e9-69d0-43ed-91b2-9d553fb555d6">

Is this what you were imagining?

Reply:
I like it. It's very clear. I wasn't happy with compare to period earlier anyways.

Reply:
https://github.com/PostHog/posthog/assets/1855120/245a3817-b1d7-4c98-8757-45cf49ec15c2

let me know what you think of this

Reply:
This is pretty great